src/components/PopoverMenu.tsx

```diff
@@ -328,7 +339,7 @@ PopoverMenu.displayName = 'PopoverMenu';
 export default React.memo(
     PopoverMenu,
     (prevProps, nextProps) =>
-        prevProps.menuItems.length === nextProps.menuItems.length &&
+        lodashIsEqual(prevProps.menuItems, nextProps.menuItems) &&
```

Comment:
Hey 👋 , coming from [this issue](https://github.com/Expensify/App/issues/51016). We changed the `menuItems.length` comparison to a deep comparison of `menuItems`, which may be causing the Popover menu flicker by re-rendering the component. We fixed it by using the existing `shouldFreezeCapture` prop in the `ActiveHoverable` component. However, if we use only the length comparison instead of a deep comparison, the issue doesn’t seem to occur

Reply:
Using the length is not enough to check if the items changed. I'm not sure if the freeze approach is best here. I think it's better to investigate why it did re-render, if the list was not changed you should memoise it.